assets/js/modules/analytics-4/utils/withCustomDimensions.js

```diff
@@ -228,11 +220,57 @@ export default function withCustomDimensions( options = {} ) {
 					MODULES_ANALYTICS_4
 				).areCustomDimensionsGatheringData( customDimensions );
 			} );
+			const dataAvailabilityReportErrors = useSelect( ( select ) => {
+				if ( ! customDimensions ) {
+					return {};
+				}
 
-			const commonErrorProps = {
-				headerText: tileTitle,
-				infoTooltip: tileInfoTooltip,
-			};
+				return select(
+					MODULES_ANALYTICS_4
+				).getDataAvailabilityReportErrors( customDimensions );
+			} );
+
+			const hasInvalidCustomDimensionError =
+				( isGatheringData &&
+					Object.values( dataAvailabilityReportErrors ).some(
+						( error ) => isInvalidCustomDimensionError( error )
+					) ) ||
+				( ! isGatheringData &&
+					isInvalidCustomDimensionError( reportError ) );
+
+			const invalidCustomDimensionReportOptions = useSelect(
+				( select ) => {
+					if ( ! hasInvalidCustomDimensionError ) {
+						return [];
+					}
+
+					if ( isGatheringData ) {
+						const { getDataAvailabilityReportOptions } =
+							select( MODULES_ANALYTICS_4 );
+
+						return Object.keys( dataAvailabilityReportErrors )
+							.filter( ( dimension ) =>
+								isInvalidCustomDimensionError(
+									dataAvailabilityReportErrors[ dimension ]
+								)
+							)
+							.map( ( dimension ) =>
+								getDataAvailabilityReportOptions( dimension )
```

Comment:
If we have the error object, we can reverse lookup its report args via `getMetaDataForError` by the error itself, so maybe this selector isn't necessary?

Reply:
That is correct, however, the `getDataAvailabilityReportErrors` selector uses the `getDataAvailabilityReportOptions` selector under the hood, so we aren't able to remove it. Is it okay to keep using `getDataAvailabilityReportOptions` here in that case?